### PR TITLE
chore(TUI-360): use react.lazy to load react ace

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

react ace is quite big and is in the main bundle
screenshot 2019-02-26 at 12 41 02

**What is the chosen solution to this problem?**

read and follow https://reactjs.org/docs/code-splitting.html#reactlazy

screenshot 2019-02-26 at 12 38 30

rework example:
https://youtu.be/RxesjUT7TTs

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/forms/src/widgets/CodeWidget/CodeWidget.component.js [hunk 1]
```diff
@@ -1,50 +1,53 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import TextareaWidget from 'react-jsonschema-form/lib/components/widgets/TextareaWidget';
```
Comment (jmfrancois) [inline packages/forms/src/widgets/CodeWidget/CodeWidget.component.js]: Note this is the OLD widget from mozilla form libs which is not anymore used and will be removed in 3.0

File: packages/forms/package.json [hunk 2]
```diff
@@ -36,11 +36,15 @@
   "dependencies": {
     "@talend/json-schema-form-core": "1.0.0-alpha.4",
     "ajv": "^6.2.1",
+    "brace": "0.11.1",
     "classnames": "^2.2.5",
+    "copy-webpack-plugin": "^5.0.4",
```
Comment (jmfrancois) [inline packages/forms/package.json]: this deps is here to expose the webpack config needed for the Code Component.

File: packages/forms/src/UIForm/fields/Code/README.md [hunk 3]
```diff
@@ -51,3 +51,22 @@ This widget allows you to render an ace-editor within a form.
...
+	// ... your current config
+	plugins: [].concat(
+		// ... your plugins
+		form.plugins,
+	),
+};
+```
+
+We need this configuration to let ace fetch its own plugin.
+To better understand why what happens you can read the source: https://github.com/thlorenz/brace/blob/master/index.js#L4115
```
Comment (jlame) [inline packages/forms/src/UIForm/fields/Code/README.md]: typo

**Comment by jlame** [comment]
we might have some issue with dynamic import and jest test, need to investigate on the issue

File: packages/forms/src/deprecated/widgets/CodeWidget/CodeWidget.test.js [hunk 4]
```diff
@@ -4,29 +4,29 @@ import { shallow, mount } from 'enzyme';
 import CodeWidget from './CodeWidget.component';
 
 describe('CodeWidget', () => {
-	it('should be AceCodeWidget', () => {
+	it('should be Code', () => {
 		expect(CodeWidget.displayName).toBe('AceCodeWidget');
 	});
 
-	it('should render ReactAce', () => {
+	xit('should render ReactAce', () => {
```
Comment (jlame) [inline packages/forms/src/deprecated/widgets/CodeWidget/CodeWidget.test.js]: test are disabled here ?

**Pull request closed without merging**